From: QbProg <tholag@gmail.com>
Date: Thu, 12 May 2011 20:43:44 +0200
Subject: [PATCH 46/82] [warning-fix][uninitialized-var]

Fixes warnings like : (potentially) uninitialized local variable 'name' used
---
 src/CSLib/CSLib.cxx                         |    2 +-
 src/Convert/Convert_ConicToBSplineCurve.cxx |    4 ++--
 src/OSD/OSD_File.cxx                        |   16 ++++++++--------
 src/OSD/OSD_FileNode.cxx                    |    4 ++--
 src/OSD/OSD_Process.cxx                     |    2 +-
 src/OSD/OSD_WNT.cxx                         |    4 ++--
 src/OSD/OSD_WNT_1.cxx                       |    6 +++---
 src/OSD/OSD_signal_WNT.cxx                  |    4 ++--
 src/Storage/Storage_Schema.cxx              |    2 +-
 src/math/math_DirectPolynomialRoots.cxx     |    1 +
 10 files changed, 23 insertions(+), 22 deletions(-)

From: tpaviot <tpaviot@gmail.com>
Date: Tue, 24 May 2011 06:47:15 +0200
Subject: [PATCH 47/82] [warning-fix][uninitialized-var]

Fix warning 'myVar' *may* be used uninitialized in this function
---
 inc/AppParCurves_ResolConstraint.gxx               |    2 +-
 inc/Blend_Walking_3.gxx                            |    2 +-
 inc/Contap_ContourGen_3.gxx                        |    2 +-
 inc/GProp_UFunction.gxx                            |    6 ++--
 inc/IntStart_SearchOnBoundaries_1.gxx              |    2 +-
 inc/Intf_InterferencePolygonPolyhedron.gxx         |    2 +
 src/AIS/AIS_Axis.cxx                               |    2 +-
 src/AIS/AIS_InteractiveContext.cxx                 |   18 +++++-----
 src/AIS/AIS_Shape.cxx                              |    6 ++--
 src/AIS2D/AIS2D_InteractiveContext.cxx             |   12 +++---
 src/Adaptor3d/Adaptor3d_CurveOnSurface.cxx         |    8 ++--
 src/Adaptor3d/Adaptor3d_TopolTool.cxx              |    2 +-
 src/AdvApprox/AdvApprox_SimpleApprox.cxx           |    2 +-
 src/Approx/Approx_SameParameter.cxx                |    1 +
 src/BRep/BRep_Builder.cxx                          |   10 +++---
 src/BRepBlend/BRepBlend_SurfRstLineBuilder.cxx     |    6 ++--
 src/BRepExtrema/BRepExtrema_DistanceSS.cxx         |    4 +-
 src/BRepFill/BRepFill_CompatibleWires.cxx          |    4 +-
 src/BRepFill/BRepFill_Generator.cxx                |    2 +-
 src/BRepFill/BRepFill_OffsetWire.cxx               |    4 +-
 src/BRepMesh/BRepMesh_Classifier.cxx               |    8 ++--
 src/ChFi3d/ChFi3d_Builder_C1.cxx                   |    4 +-
 src/ChFi3d/ChFi3d_Builder_CnCrn.cxx                |    4 +-
 src/ChFi3d/ChFi3d_ChBuilder.cxx                    |    1 +
 src/ChFiDS/ChFiDS_FilSpine.cxx                     |    4 +-
 src/Draw/Draw_Viewer.cxx                           |    4 +-
 src/GeomFill/GeomFill_GuideTrihedronPlan.cxx       |    2 +-
 src/Graphic3d/Graphic3d_ArrayOfPrimitives.cxx      |    2 +-
 src/HLRBRep/HLRBRep_ShapeToHLR.cxx                 |    2 +-
 src/IntPatch/IntPatch_ALineToWLine.cxx             |    2 +-
 src/IntPatch/IntPatch_ImpPrmIntersection.cxx       |    2 +-
 src/IntPatch/IntPatch_PrmPrmIntersection.cxx       |    2 +
 src/Interface/Interface_FloatWriter.cxx            |    2 +-
 src/LDOM/LDOM_XmlReader.cxx                        |    2 +-
 src/MoniTool/MoniTool_TypedValue.cxx               |    6 ++--
 src/OpenGl/OpenGl_TextureBox.cxx                   |    2 +-
 src/ProjLib/ProjLib_CompProjectedCurve.cxx         |    2 +-
 src/ProjLib/ProjLib_ProjectedCurve.cxx             |    2 +-
 src/Quantity/Quantity_Color.cxx                    |    2 +-
 ...oTolAndGeoTolWthDatRefAndModGeoTolAndPosTol.cxx |    2 +-
 .../RWStepDimTol_RWModifiedGeometricTolerance.cxx  |    2 +-
 .../RWStepElement_RWCurve3dElementDescriptor.cxx   |    2 +-
 .../RWStepElement_RWElementDescriptor.cxx          |    2 +-
 .../RWStepElement_RWSurface3dElementDescriptor.cxx |    4 +-
 .../RWStepElement_RWVolume3dElementDescriptor.cxx  |    4 +-
 src/RWStepFEA/RWStepFEA_RWFeaAxis2Placement3d.cxx  |    2 +-
 src/ShapeAnalysis/ShapeAnalysis_CheckSmallFace.cxx |    4 +-
 src/ShapeAnalysis/ShapeAnalysis_ShapeTolerance.cxx |    2 +-
 .../ShapeConstruct_ProjectCurveOnSurface.cxx       |    1 +
 src/TopOpeBRep/TopOpeBRep_FFTransitionTool.cxx     |    4 +-
 src/TopOpeBRepBuild/TopOpeBRepBuild_GIter.cxx      |    2 +-
 src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx      |    2 +-
 src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx        |    8 +++--
 src/TopOpeBRepDS/TopOpeBRepDS_EXPORT.cxx           |   38 ++++++++++----------
 .../TopOpeBRepDS_Edge3dInterferenceTool.cxx        |    3 +-
 src/TopOpeBRepDS/TopOpeBRepDS_HDataStructure.cxx   |    2 +-
 src/Viewer2dTest/Viewer2dTest_DisplayCommands.cxx  |    2 +-
 src/VrmlData/VrmlData_Material.cxx                 |    2 +-
 58 files changed, 125 insertions(+), 115 deletions(-)

From: tpaviot <tpaviot@gmail.com>
Date: Tue, 24 May 2011 08:24:39 +0200
Subject: [PATCH 48/82] [warning-fix][uninitialized-var]

Fix warning 'myVar' *is* used uninitialized in this function
---
 src/Adaptor3d/Adaptor3d_CurveOnSurface.cxx      |    2 +-
 src/IntCurve/IntCurve_IntConicConic_Tool.cxx    |    6 +++---
 src/Prs3d/Prs3d_TextAspect.cxx                  |    6 +-----
 src/XmlMDataStd/XmlMDataStd_NamedDataDriver.cxx |   10 +++++-----
 4 files changed, 10 insertions(+), 14 deletions(-)

commit f605499b74be6931c7805a6d0ea5a9149a7fb964
Author: QbProg <tholag@gmail.com>
Date:   Mon Jun 13 20:20:00 2011 +0200

    Another small fix

commit 3ef990833fe7d6dd126ef0fc334d8e9c23227083
Author: QbProg <tholag@gmail.com>
Date:   Mon Jun 13 20:09:46 2011 +0200

    Forgot to revert a pourSGI change

commit fbdee817a9c1cced72a5fec6d1aef91deb436ee7
Author: QbProg <tholag@gmail.com>
Date:   Mon Jun 13 19:44:19 2011 +0200

    Fixed pourSGI improper initialization (causing different behavoir from mainstream)

From: QbProg <tholag@gmail.com>
Date: Sat, 11 Jun 2011 16:38:23 +0200
Subject: [PATCH 49/82] [warning-fix][uninitialized-var]

Removed unuinitialized vars from TKIGES
---
 src/IGESAppli/IGESAppli_ToolLineWidening.cxx    |    2 +-
 src/IGESData/IGESData_ParamReader.cxx           |    8 ++++----
 src/IGESGeom/IGESGeom_ToolBSplineSurface.cxx    |    2 +-
 src/IGESGeom/IGESGeom_ToolConicArc.cxx          |    2 +-
 src/IGESGeom/IGESGeom_ToolPlane.cxx             |    2 +-
 src/IGESGraph/IGESGraph_ToolUniformRectGrid.cxx |    4 ++--
 src/IGESToBRep/IGESToBRep_TopoSurface.cxx       |    2 +-
 7 files changed, 11 insertions(+), 11 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sat, 11 Jun 2011 16:59:55 +0200
Subject: [PATCH 50/82] [warning-fix][uninitialized-var]

Fixes (and todos) in TKStepBase
---
 src/RWStepBasic/RWStepBasic_RWSiUnit.cxx           |    6 ++++--
 .../RWStepBasic_RWSiUnitAndAreaUnit.cxx            |    6 ++++--
 .../RWStepBasic_RWSiUnitAndLengthUnit.cxx          |    6 ++++--
 .../RWStepBasic_RWSiUnitAndMassUnit.cxx            |    6 ++++--
 .../RWStepBasic_RWSiUnitAndPlaneAngleUnit.cxx      |    6 ++++--
 .../RWStepBasic_RWSiUnitAndRatioUnit.cxx           |    6 ++++--
 .../RWStepBasic_RWSiUnitAndSolidAngleUnit.cxx      |    8 +++++---
 ...sic_RWSiUnitAndThermodynamicTemperatureUnit.cxx |    6 ++++--
 .../RWStepBasic_RWSiUnitAndTimeUnit.cxx            |    6 ++++--
 .../RWStepBasic_RWSiUnitAndVolumeUnit.cxx          |    6 ++++--
 10 files changed, 41 insertions(+), 21 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sat, 11 Jun 2011 19:40:58 +0200
Subject: [PATCH 51/82] [warning-fix][uninitialized-var]

Fixes many potential uninitialized vars around Draw and Test
---
 src/BRepTest/BRepTest_CurveCommands.cxx      |    2 +-
 src/DNaming/DNaming_TransformationDriver.cxx |    6 +++---
 src/Draw/Draw.cxx                            |    2 +-
 src/Draw/Draw_GraphicCommands.cxx            |    2 +-
 src/Draw/MainWindow.cxx                      |    2 +-
 src/DrawTrSurf/DrawTrSurf.cxx                |    2 +-
 src/SWDRAW/SWDRAW_ShapeTool.cxx              |    2 +-
 src/ViewerTest/ViewerTest_ViewerCommands.cxx |    2 +-
 src/VrmlData/VrmlData_Node.cxx               |    2 +-
 src/VrmlData/VrmlData_Scene.cxx              |    4 ++--
 src/XCAFPrs/XCAFPrs_AISObject.cxx            |    8 ++++----
 11 files changed, 17 insertions(+), 17 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sun, 12 Jun 2011 12:03:06 +0200
Subject: [PATCH 52/82] [warning-fix][uninitialized-var]

Fixed many potentially un-init vars
---
 inc/Approx_ComputeCLine.gxx                        |    5 +++--
 inc/Blend_Walking_1.gxx                            |    4 ++--
 inc/Blend_Walking_3.gxx                            |    2 +-
 inc/Blend_Walking_4.gxx                            |    6 +++---
 inc/IntCurveSurface_Inter.gxx                      |    2 +-
 inc/IntPatch_ImpImpIntersection_0.gxx              |    2 +-
 inc/IntStart_SearchOnBoundaries_1.gxx              |    2 +-
 inc/IntWalk_IWalking_4.gxx                         |    2 +-
 src/BRepBlend/BRepBlend_RstRstLineBuilder.cxx      |    4 ++--
 src/BRepBlend/BRepBlend_SurfRstLineBuilder.cxx     |    8 ++++----
 src/BRepFeat/BRepFeat_Form.cxx                     |    4 ++--
 src/BRepFill/BRepFill_EdgeOnSurfLaw.cxx            |    2 +-
 src/BRepFill/BRepFill_Evolved.cxx                  |    2 +-
 src/BRepFill/BRepFill_Generator.cxx                |    2 +-
 src/BRepFill/BRepFill_MultiLine.cxx                |    2 +-
 src/BRepFill/BRepFill_SectionPlacement.cxx         |    2 +-
 src/ChFi3d/ChFi3d_Builder_0.cxx                    |    6 +++---
 src/ChFi3d/ChFi3d_Builder_2.cxx                    |    2 +-
 src/ChFi3d/ChFi3d_Builder_6.cxx                    |    4 ++--
 src/ChFi3d/ChFi3d_Builder_C1.cxx                   |    4 ++--
 src/ChFi3d/ChFi3d_Builder_C2.cxx                   |    2 +-
 src/ChFi3d/ChFi3d_Builder_CnCrn.cxx                |   16 ++++++++--------
 src/ChFi3d/ChFi3d_FilBuilder_C2.cxx                |    2 +-
 src/GeomFill/GeomFill_CorrectedFrenet.cxx          |    2 +-
 src/GeomFill/GeomFill_TgtOnCoons.cxx               |    6 +++---
 src/IntCurve/IntCurve_IntConicConic_1.cxx          |    2 +-
 src/IntPatch/IntPatch_ALineToWLine.cxx             |    2 +-
 src/Law/Law_BSpFunc.cxx                            |    6 +++---
 src/LocOpe/LocOpe_Generator.cxx                    |    2 +-
 src/TopOpeBRep/TopOpeBRep_FacesFiller_1.cxx        |    2 +-
 src/TopOpeBRep/TopOpeBRep_kpart.cxx                |    2 +-
 src/TopOpeBRep/TopOpeBRep_vprdeg.cxx               |    2 +-
 src/TopOpeBRepBuild/TopOpeBRepBuild_BuilderON.cxx  |    6 +++---
 .../TopOpeBRepBuild_BuilderON2d.cxx                |    4 ++--
 src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx        |    2 +-
 src/TopOpeBRepDS/TopOpeBRepDS_EXPORT.cxx           |   12 ++++++------
 .../TopOpeBRepDS_ProcessEdgeInterferences.cxx      |    2 +-
 src/TopOpeBRepDS/TopOpeBRepDS_redu.cxx             |    2 +-
 38 files changed, 71 insertions(+), 70 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Fri, 17 Jun 2011 21:03:45 +0200
Subject: [PATCH 53/82] [warning-fix][uninitialized-var]

Fixed uninitialized variables
---
 inc/AppParCurves_Variational_1.gxx    |    2 +-
 inc/GCPnts_TangentialDeflection.gxx   |    2 +-
 src/AdvApp2Var/AdvApp2Var_Iso.cxx     |    2 +-
 src/GeomLib/GeomLib.cxx               |    6 +++---
 src/IntAna/IntAna_Curve.cxx           |    4 ++--
 src/ProjLib/ProjLib_ComputeApprox.cxx |    4 ++--
 6 files changed, 10 insertions(+), 10 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Fri, 17 Jun 2011 21:15:23 +0200
Subject: [PATCH 54/82] [warning-fix][uninitialized-var]

fixed uninitialized vars warnings
---
 .../ProjLib_ComputeApproxOnPolarSurface.cxx        |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sun, 19 Jun 2011 17:52:15 +0200
Subject: [PATCH 55/82] [warning-fix][uninitialized-var]

Fixes warnings like : (potentially) uninitialized local variable 'name' used
---
 inc/Contap_ContourGen_2.gxx                        |    2 +-
 inc/Contap_ContourGen_3.gxx                        |    4 +-
 src/AIS/AIS_DiameterDimension.cxx                  |    2 +-
 src/AIS/AIS_TexturedShape.cxx                      |    4 +-
 src/Aspect/Aspect_MarkMapEntry.cxx                 |    2 +-
 src/BOPTools/BOPTools_Tools3D.cxx                  |    2 +-
 src/BRepMesh/BRepMesh_IncrementalMesh.cxx          |    2 +-
 src/BRepOffset/BRepOffset_Inter3d.cxx              |    4 +-
 src/BRepTools/BRepTools_ShapeSet.cxx               |    2 +-
 src/BinTools/BinTools_ShapeSet.cxx                 |    4 +-
 src/CGM/cgmochar.c                                 |    2 +-
 src/Draft/Draft_Modification_1.cxx                 |    4 +-
 src/Geom/Geom_OffsetSurface.cxx                    |    2 +-
 src/Geom2dAdaptor/Geom2dAdaptor_Curve.cxx          |   12 +++---
 src/GeomAdaptor/GeomAdaptor_Curve.cxx              |   12 +++---
 src/HLRAlgo/HLRAlgo_PolyInternalData.cxx           |    4 +-
 src/HLRAlgo/HLRAlgo_PolyShellData.cxx              |    2 +-
 src/HLRBRep/HLRBRep_PolyAlgo.cxx                   |    2 +-
 src/IntTools/IntTools_EdgeFace.cxx                 |    2 +-
 src/IntTools/IntTools_TopolTool.cxx                |    2 +-
 src/MAT2d/MAT2d_Tool2d.cxx                         |    2 +-
 src/MeshVS/MeshVS_MeshPrsBuilder.cxx               |    4 +-
 src/Select3D/Select3D_SensitiveFace.cxx            |    2 +-
 src/ShapeAnalysis/ShapeAnalysis_CheckSmallFace.cxx |    2 +-
 src/ShapeAnalysis/ShapeAnalysis_Surface.cxx        |    2 +-
 .../ShapeAnalysis_TransferParameters.cxx           |    2 +-
 src/ShapeAnalysis/ShapeAnalysis_WireOrder.cxx      |    2 +-
 src/ShapeConstruct/ShapeConstruct.cxx              |    2 +-
 src/ShapeCustom/ShapeCustom_Surface.cxx            |    2 +-
 src/ShapeFix/ShapeFix_ComposeShell.cxx             |    2 +-
 src/ShapeFix/ShapeFix_Wire.cxx                     |   10 +++---
 src/ShapeUpgrade/ShapeUpgrade_ClosedEdgeDivide.cxx |    2 +-
 src/ShapeUpgrade/ShapeUpgrade_EdgeDivide.cxx       |    2 +-
 .../ShapeUpgrade_SplitSurfaceAngle.cxx             |    2 +-
 src/ShapeUpgrade/ShapeUpgrade_WireDivide.cxx       |    2 +-
 src/V3d/V3d_Camera.cxx                             |    4 +-
 src/WNT/EHDC.cxx                                   |    4 +-
 src/WNT/WNT_IconBox.cxx                            |    2 +-
 src/WNT/WNT_IconBox_1.cxx                          |    6 ++--
 src/WNT/WNT_ImageManager.cxx                       |    2 +-
 src/WNT/WNT_ImageProcessor.cxx                     |   34 ++++++++++----------
 src/WNT/WNT_MFTDraw.cxx                            |    2 +-
 src/WNT/WNT_WDriver.cxx                            |    2 +-
 src/WNT/WNT_Window.cxx                             |    4 +-
 src/WNT/WNT_WndProc.cxx                            |    2 +-
 45 files changed, 87 insertions(+), 87 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Sun, 19 Jun 2011 18:20:52 +0200
Subject: [PATCH 56/82] [warning-fix][uninitialized-var] Fixes warnings like :
 (potentially) uninitialized local variable 'name'
 used

---
 src/Visual3d/Visual3d_View.cxx |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

From: QbProg <tholag@gmail.com>
Date: Wed, 22 Jun 2011 19:43:20 +0200
Subject: [PATCH 57/82] [warning-fix][uninitialized-var]

Fixed annoying runtime error appearing in MSVC/WIN32 about an uninitialized var
---
 src/OSD/OSD_Environment.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

From: tpaviot <tpaviot@gmail.com>
Date: Tue, 28 Jun 2011 11:10:08 +0200
Subject: [PATCH 58/82] [warning-fix][uninitialized-var] Fix warnings: *
 oce/src/Storage/Storage_Schema.cxx:451:0
 'bit.Storage_BucketIterator::myBucket' may be used
 uninitialized in this function

* oce/src/Storage/Storage_Schema.cxx:451:0
'bit.Storage_BucketIterator::myCurrentBucket' may be used uninitialized
in this function

* oce/src/Storage/Storage_Schema.cxx:451:0
'bit.Storage_BucketIterator::myBucketNumber' may be used uninitialized
in this function
---
 src/Storage/Storage_Schema.cxx |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

From 74b613cfaddaee006817ec84ea5e46249de91bc1 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Tue, 19 Jul 2011 11:39:29 +0200
Subject: [PATCH 02/26] [warning-fix][uninitialized-var] Fixed potential
 uninitialized variable in NIS_View.cxx

---
 src/NIS/NIS_View.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

From 84bcb8037b609f304796ddaa3c5651d7af8d5fe9 Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sun, 24 Jul 2011 18:22:50 +0200
Subject: [PATCH 10/11] [warning-fix][uninitialized-var] Fixes warnings like :
 (potentially) uninitialized local variable 'name'
 used

---
 src/OSD/OSD_Process.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: oce/src/AIS/AIS_InteractiveContext.cxx
===================================================================
--- oce.orig/src/AIS/AIS_InteractiveContext.cxx
+++ oce/src/AIS/AIS_InteractiveContext.cxx
@@ -547,7 +547,7 @@ void AIS_InteractiveContext::Load(const
 
   if(!HasOpenedContext()) {
     if(SelMode==-1 && !AllowDecomposition){
-      Standard_Integer DispMode,HiMod,SelModeDef;
+      Standard_Integer DispMode = 0,HiMod,SelModeDef = 0;
       
       GetDefModes(anIObj,DispMode,HiMod,SelModeDef);
       Handle(AIS_GlobalStatus) STATUS= 
@@ -741,7 +741,7 @@ Standard_Boolean AIS_InteractiveContext:
   
   
   if(LS->IsTemporary()){
-    Standard_Integer DM,HM,SM;
+    Standard_Integer DM = 0,HM,SM = 0;
     GetDefModes(anIObj,DM,HM,SM);
     
     SM = LS->SelectionModes().IsEmpty() ? SM : LS->SelectionModes().First();
@@ -1563,7 +1563,7 @@ void AIS_InteractiveContext::SetDisplayM
           STATUS->AddDisplayMode(aMode);
 
         myMainPM->Display(anIObj,aMode);
-        Standard_Integer DM,HM,SM;
+        Standard_Integer DM,HM = 0,SM;
         GetDefModes(anIObj,DM,HM,SM);
         if(STATUS->IsHilighted()){
           myMainPM->Highlight(anIObj,HM);
@@ -1608,7 +1608,7 @@ UnsetDisplayMode(const Handle(AIS_Intera
           myMainPM->Unhighlight(anIObj,OldMode);
         myMainPM->SetVisibility (anIObj, OldMode, Standard_False);
         myMainPM->Display(anIObj,myDisplayMode);
-        Standard_Integer DM,HM,SM;
+        Standard_Integer DM,HM = 0,SM;
         GetDefModes(anIObj,DM,HM,SM);
         if(STATUS->IsHilighted()){
           myMainPM->Highlight(anIObj,HM);
Index: oce/src/AIS/AIS_Shape.cxx
===================================================================
--- oce.orig/src/AIS/AIS_Shape.cxx
+++ oce/src/AIS/AIS_Shape.cxx
@@ -217,8 +217,8 @@ void AIS_Shape::Compute(const Handle(Prs
     {
       Standard_Real prevangle ;
       Standard_Real newangle  ; 
-      Standard_Real prevcoeff ;
-      Standard_Real newcoeff  ; 
+      Standard_Real prevcoeff = 0;
+      Standard_Real newcoeff = 0 ; 
       
       Standard_Boolean isOwnDeviationAngle = OwnDeviationAngle(newangle,prevangle);
       Standard_Boolean isOwnDeviationCoefficient = OwnDeviationCoefficient(newcoeff,prevcoeff);
Index: oce/src/Adaptor3d/Adaptor3d_CurveOnSurface.cxx
===================================================================
--- oce.orig/src/Adaptor3d/Adaptor3d_CurveOnSurface.cxx
+++ oce/src/Adaptor3d/Adaptor3d_CurveOnSurface.cxx
@@ -194,7 +194,7 @@ static void FindBounds(const TColStd_Arr
 		       Standard_Boolean& DerNull)
 
 {
-  Standard_Integer N;
+  Standard_Integer N = 0;
   Standard_Real Tol=Precision::PConfusion()/10;
   Hunt(Arr,Coord,N);
   DerNull=Standard_False;
Index: oce/src/ChFi3d/ChFi3d_Builder_0.cxx
===================================================================
--- oce.orig/src/ChFi3d/ChFi3d_Builder_0.cxx
+++ oce/src/ChFi3d/ChFi3d_Builder_0.cxx
@@ -3918,6 +3918,7 @@ Standard_EXPORT
   Wreffin = WL;
   nwf = WF;
   nwl = WL;
+  pared = 0.0;
   nbed = Spine->NbEdges();
   periodic = Spine->IsPeriodic();
   if(periodic) {
Index: oce/src/ChFi3d/ChFi3d_Builder_CnCrn.cxx
===================================================================
--- oce.orig/src/ChFi3d/ChFi3d_Builder_CnCrn.cxx
+++ oce/src/ChFi3d/ChFi3d_Builder_CnCrn.cxx
@@ -306,7 +306,7 @@ static void CurveHermite (const TopOpeBR
   gp_Vec d11,d12;
   Standard_Integer ii,jj;
   Standard_Real up1,up2;
-  Standard_Integer ilin,jfp;
+  Standard_Integer ilin = 0,jfp = 0;
   Handle (Geom_Curve) c1,c2;
   if (sharpicmoins) {
     c1=BRep_Tool::Curve(Eviveicmoins,up1,up2);
Index: oce/src/Draw/Draw_Viewer.cxx
===================================================================
--- oce.orig/src/Draw/Draw_Viewer.cxx
+++ oce/src/Draw/Draw_Viewer.cxx
@@ -519,7 +519,7 @@ void Draw_Viewer::SetZoom (const Standar
   if (myViews[id]) {
     Standard_Real zz = z / myViews[id]->Zoom;
     myViews[id]->Zoom = z;
-    Standard_Integer X,Y,W,H;
+    Standard_Integer X,Y,W = 0,H = 0;
     GetPosSize(id,X,Y,W,H);
     Standard_Real w = W;
     myViews[id]->dX = (Standard_Integer )(   w / 2 - zz * (w/2 - myViews[id]->dX));
@@ -1822,7 +1822,7 @@ void Draw_Display::DrawTo (const gp_Pnt2
   case DRAW : {
 
 #if 1
-    Standard_Integer x0,y0,x1,y1;
+    Standard_Integer x0 = 0,y0 = 0,x1 = 0,y1 = 0;
     curview->GetFrame(x0,y0,x1,y1);
 
 
Index: oce/src/GCPnts/GCPnts_TangentialDeflection.gxx
===================================================================
--- oce.orig/src/GCPnts/GCPnts_TangentialDeflection.gxx
+++ oce/src/GCPnts/GCPnts_TangentialDeflection.gxx
@@ -247,7 +247,7 @@ void GCPnts_TangentialDeflection::Perfor
   Standard_Integer i;       
   gp_XYZ V1, V2;
   gp_Pnt MiddlePoint, CurrentPoint, LastPoint;   
-  Standard_Real Du, Dusave, MiddleU, L1, L2;
+  Standard_Real Du, Dusave, MiddleU = 0.0, L1 = 0.0, L2 = 0.0;
 
   Standard_Real U1       = firstu;   
   Standard_Real LTol     = Precision::Confusion ();  //protection longueur nulle
Index: oce/src/GProp/GProp_UFunction.gxx
===================================================================
--- oce.orig/src/GProp/GProp_UFunction.gxx
+++ oce/src/GProp/GProp_UFunction.gxx
@@ -109,7 +109,7 @@ Standard_Boolean GProp_UFunction::Center
 {
   gp_XYZ        aPmP0;
   Standard_Real aS;
-  Standard_Real aD1;
+  Standard_Real aD1 = 0;
 
   F = VolumeValue(X, aPmP0, aS, aD1);
 
@@ -149,8 +149,8 @@ Standard_Boolean GProp_UFunction::Inerti
 					             Standard_Real &F)
 {
   gp_XYZ        aPmP0;
-  Standard_Real aS;
-  Standard_Real aD1;
+  Standard_Real aS = 0;
+  Standard_Real aD1 = 0;
   Standard_Real aParam1;
   Standard_Real aParam2;
   Standard_Real *aCoeffs = (Standard_Real *)myCoeffs;
Index: oce/src/IntPatch/IntPatch_PrmPrmIntersection.cxx
===================================================================
--- oce.orig/src/IntPatch/IntPatch_PrmPrmIntersection.cxx
+++ oce/src/IntPatch/IntPatch_PrmPrmIntersection.cxx
@@ -3121,6 +3121,8 @@ void IntPatch_PrmPrmIntersection::PointD
 	//-- dans chacun des tableaux 
 		  Standard_Real Dist3[3],u3[3] = { 0.0, 0.0, 0.0 },v3[3] = { 0.0, 0.0, 0.0 };
 	Dist3[0]=Dist3[1]=Dist3[2]=RealLast();
+	u3[0]=u3[1]=u3[2]=0;
+	v3[0]=v3[1]=v3[2]=0;
 	for(U=resu0,i=0; i<SU1; i++,U+=du1) { 
 	  for(V=resv0,j=0; j<SV1; V+=dv1,j++) {       
 	    //-- On place les 3 meilleures valeurs dans Dist1,Dist2,Dist3
Index: oce/src/MoniTool/MoniTool_TypedValue.cxx
===================================================================
--- oce.orig/src/MoniTool/MoniTool_TypedValue.cxx
+++ oce/src/MoniTool/MoniTool_TypedValue.cxx
@@ -178,7 +178,7 @@ static Standard_Boolean StaticPath(const
       break;
     case MoniTool_ValueEnum    : {
       def.AssignCat("Enum");
-      Standard_Integer startcase, endcase; Standard_Boolean match;
+      Standard_Integer startcase = 0, endcase = 0; Standard_Boolean match = Standard_False;
       EnumDef (startcase,endcase,match);
       Sprintf(mess," [%s%d-%d]",(match ? "in " : ""),startcase,endcase);
       def.AssignCat(mess);
@@ -598,8 +598,8 @@ static Standard_Boolean StaticPath(const
     }
     case MoniTool_ValueEnum    : {
 //  On admet les deux formes : Enum de preference, sinon Integer
-      Standard_Integer startcase, endcase;// unused ival;
-      Standard_Boolean match;
+      Standard_Integer startcase = 0, endcase;// unused ival;
+      Standard_Boolean match = 0;
       EnumDef (startcase,endcase,match);
       if (!match) return Standard_True;
       if (EnumCase (val->ToCString()) >= startcase) return Standard_True;
Index: oce/src/Quantity/Quantity_Color.cxx
===================================================================
--- oce.orig/src/Quantity/Quantity_Color.cxx
+++ oce/src/Quantity/Quantity_Color.cxx
@@ -408,7 +408,7 @@ Standard_ShortReal HH, LL, SS;
 
 void Quantity_Color::HlsRgb (const Quantity_Parameter H, const Quantity_Parameter L, const Quantity_Parameter S, Quantity_Parameter& R, Quantity_Parameter& G, Quantity_Parameter& B) {
 
-Standard_ShortReal RR, GG, BB;
+Standard_ShortReal RR=0, GG=0, BB=0;
 
 	call_hlsrgb (Standard_ShortReal (H),
 			Standard_ShortReal (L),
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnit.cxx
@@ -101,6 +101,8 @@ void RWStepBasic_RWSiUnit::ReadStep(cons
   else
     ach->AddFail("Parameter #3 (name) is not an enumeration");
 
+  // @todo Apart the fail, nothing is done , and wrong enum values are used
+
   //--- Initialisation of the read entity ---
   ent->Init(hasAprefix, aPrefix, aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndAreaUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndAreaUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndAreaUnit.cxx
@@ -59,7 +59,7 @@ void RWStepBasic_RWSiUnitAndAreaUnit::Re
     }
   }
      
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -72,6 +72,8 @@ void RWStepBasic_RWSiUnitAndAreaUnit::Re
     return;
   }
   
+  // @todo Apart the fail, nothing is done , and wrong enum values are used
+ 
   ent->Init(hasAprefix,aPrefix,aName);
   ent->SetDimensions(aDimensions);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndLengthUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndLengthUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndLengthUnit.cxx
@@ -74,7 +74,7 @@ void RWStepBasic_RWSiUnitAndLengthUnit::
   } 
   
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -87,6 +87,8 @@ void RWStepBasic_RWSiUnitAndLengthUnit::
     return;
   }
 
+  // @todo Apart the fail, nothing is done , and wrong enum values are used
+
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndMassUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndMassUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndMassUnit.cxx
@@ -83,7 +83,7 @@ void RWStepBasic_RWSiUnitAndMassUnit::Re
   } 
   
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -96,6 +96,8 @@ void RWStepBasic_RWSiUnitAndMassUnit::Re
     return;
   }
 
+   // @todo Apart the fail, nothing is done , and wrong enum values are used
+
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndPlaneAngleUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndPlaneAngleUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndPlaneAngleUnit.cxx
@@ -67,7 +67,7 @@ void RWStepBasic_RWSiUnitAndPlaneAngleUn
   }
 	
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -80,6 +80,8 @@ void RWStepBasic_RWSiUnitAndPlaneAngleUn
     return;
   }
   
+ // @todo Apart the fail, nothing is done , and wrong enum values are used
+
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndRatioUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndRatioUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndRatioUnit.cxx
@@ -66,7 +66,7 @@ void RWStepBasic_RWSiUnitAndRatioUnit::R
   }
 
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -79,6 +79,8 @@ void RWStepBasic_RWSiUnitAndRatioUnit::R
     return;
   }
   
+   // @todo Apart the fail, nothing is done , and wrong enum values are used
+ 
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndSolidAngleUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndSolidAngleUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndSolidAngleUnit.cxx
@@ -63,7 +63,7 @@ void RWStepBasic_RWSiUnitAndSolidAngleUn
   }
   
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -79,7 +79,9 @@ void RWStepBasic_RWSiUnitAndSolidAngleUn
   // --- Instance of plex componant SolidAngleUnit ---
   num = data->NextForComplex(num);
   if (!data->CheckNbParams(num,0,ach,"solid_angle_unit")) return;
-
+ 
+  // @todo Apart the fail, nothing is done , and wrong enum values are used
+ 
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndThermodynamicTemperatureUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndThermodynamicTemperatureUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndThermodynamicTemperatureUnit.cxx
@@ -75,7 +75,7 @@ void RWStepBasic_RWSiUnitAndThermodynami
   }
   
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -92,6 +92,8 @@ void RWStepBasic_RWSiUnitAndThermodynami
   num = data->NextForComplex(num);
   if (!data->CheckNbParams(num,0,ach,"thermodynamic_temperature_unit")) return;
 
+   // @todo Apart the fail, nothing is done , and wrong enum values are used
+ 
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndTimeUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndTimeUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndTimeUnit.cxx
@@ -74,7 +74,7 @@ void RWStepBasic_RWSiUnitAndTimeUnit::Re
   }
   
   // --- field : name ---
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -91,6 +91,8 @@ void RWStepBasic_RWSiUnitAndTimeUnit::Re
   num = data->NextForComplex(num);
   if (!data->CheckNbParams(num,0,ach,"time_unit")) return;
 
+   // @todo Apart the fail, nothing is done , and wrong enum values are used
+ 
   //--- Initialisation of the red entity ---
   ent->Init(hasAprefix,aPrefix,aName);
 }
Index: oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndVolumeUnit.cxx
===================================================================
--- oce.orig/src/RWStepBasic/RWStepBasic_RWSiUnitAndVolumeUnit.cxx
+++ oce/src/RWStepBasic/RWStepBasic_RWSiUnitAndVolumeUnit.cxx
@@ -54,7 +54,7 @@ void RWStepBasic_RWSiUnitAndVolumeUnit::
     }
   }
      
-  StepBasic_SiUnitName aName;
+  StepBasic_SiUnitName aName = StepBasic_sunMetre; // 0
   if (data->ParamType(num,2) == Interface_ParamEnum) {
     Standard_CString text = data->ParamCValue(num,2);
     if(!reader.DecodeName(aName,text)){
@@ -70,6 +70,8 @@ void RWStepBasic_RWSiUnitAndVolumeUnit::
   data->NamedForComplex("VOLUME_UNIT",num0,num,ach);
   if (!data->CheckNbParams(num,0,ach,"volume_unit")) return;
   
+   // @todo Apart the fail, nothing is done , and wrong enum values are used
+ 
   ent->Init(hasAprefix,aPrefix,aName);
   ent->SetDimensions(aDimensions);
 }
Index: oce/src/Storage/Storage_Schema.cxx
===================================================================
--- oce.orig/src/Storage/Storage_Schema.cxx
+++ oce/src/Storage/Storage_Schema.cxx
@@ -218,7 +218,12 @@ Storage_BucketIterator::Storage_BucketIt
     myCurrentIndex       = 0;
     myMoreObject         = Standard_True;
   }
-  else myMoreObject         = Standard_False;
+  else {
+    myBucket             = NULL;
+    myCurrentBucket      = 0;
+    myBucketNumber       = 0;
+    myMoreObject         = Standard_False;
+  }
 }
 
 //=======================================================================
Index: oce/src/TopOpeBRepBuild/TopOpeBRepBuild_BuilderON.cxx
===================================================================
--- oce.orig/src/TopOpeBRepBuild/TopOpeBRepBuild_BuilderON.cxx
+++ oce/src/TopOpeBRepBuild/TopOpeBRepBuild_BuilderON.cxx
@@ -1526,7 +1526,7 @@ void TopOpeBRepBuild_BuilderON::GFillONP
 //    Standard_Boolean samerk = (rankEG == rkToFill);
 
 //    TopAbs_Orientation oegFOR;
-    Standard_Boolean shareG;
+    Standard_Boolean shareG = false;
     Standard_Boolean ok = FUN_ds_shareG(myPB->DataStructure(),iFOR,iFCX,GI,
 			   TopoDS::Edge(EspON),shareG);
     if (!ok) return; // nyiFUNRAISE
Index: oce/src/TopOpeBRepBuild/TopOpeBRepBuild_GIter.cxx
===================================================================
--- oce.orig/src/TopOpeBRepBuild/TopOpeBRepBuild_GIter.cxx
+++ oce/src/TopOpeBRepBuild/TopOpeBRepBuild_GIter.cxx
@@ -77,7 +77,7 @@ void TopOpeBRepBuild_GIter::Current(TopA
 void TopOpeBRepBuild_GIter::Dump(Standard_OStream& OS) const 
 {
   if ( !More()) return;
-  TopAbs_State s1,s2; Current(s1,s2);
+  TopAbs_State s1 = TopAbs_UNKNOWN,s2 = TopAbs_UNKNOWN; Current(s1,s2);
   Standard_Boolean b = MYGTOPO.Value(s1,s2);
   TopOpeBRepBuild_GTopo::DumpSSB(OS,s1,s2,b); OS<<endl;
 }
Index: oce/src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx
===================================================================
--- oce.orig/src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx
+++ oce/src/TopOpeBRepBuild/TopOpeBRepBuild_GTopo.cxx
@@ -172,7 +172,7 @@ Standard_Boolean TopOpeBRepBuild_GTopo::
 
 Standard_Boolean TopOpeBRepBuild_GTopo::Value(const Standard_Integer II) const
 {
-  Standard_Integer i1,i2; Index(II,i1,i2);
+  Standard_Integer i1 = 0,i2 = 0; Index(II,i1,i2);
   Standard_Boolean b = Value(i1,i2);
   return b;
 }
Index: oce/src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx
===================================================================
--- oce.orig/src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx
+++ oce/src/TopOpeBRepDS/TopOpeBRepDS_BuildTool.cxx
@@ -601,7 +601,8 @@ void TopOpeBRepDS_BuildTool::UpdateEdgeC
 
 //  newtol *= 1.5;
 
-  TopoDS_Vertex Vmin,Vmax; Standard_Real parmin,parmax;
+  TopoDS_Vertex Vmin,Vmax;
+  Standard_Real parmin = 0,parmax = 0;
   GetOrientedEdgeVertices(E,Vmin,Vmax,parmin,parmax);
   
   Standard_Real tolmin=BRep_Tool::Tolerance(Vmin);
@@ -693,7 +694,8 @@ void  TopOpeBRepDS_BuildTool::ApproxCurv
   // Vmin,Vmax = bounding vertices of edge <E>
   // and their parameters parmin,parmax .
 
-  TopoDS_Vertex Vmin,Vmax;Standard_Real parmin,parmax;
+  TopoDS_Vertex Vmin,Vmax;
+  Standard_Real parmin = 0,parmax = 0;
   GetOrientedEdgeVertices(E,Vmin,Vmax,parmin,parmax);
 
 #ifdef DEB
@@ -1267,7 +1269,7 @@ void  TopOpeBRepDS_BuildTool::Parameter(
 					TopoDS_Shape& E,
 					TopoDS_Shape& V)const 
 {
-  Standard_Real newparam;
+  Standard_Real newparam = 0;
   Project(C.Curve(),TopoDS::Vertex(V),newparam);
   Parameter(E,V,newparam);
 }
Index: oce/src/TopOpeBRepDS/TopOpeBRepDS_EXPORT.cxx
===================================================================
--- oce.orig/src/TopOpeBRepDS/TopOpeBRepDS_EXPORT.cxx
+++ oce/src/TopOpeBRepDS/TopOpeBRepDS_EXPORT.cxx
@@ -349,7 +349,7 @@ Standard_EXPORT Standard_Boolean FDS_SIi
     TopOpeBRepDS_ListIteratorOfListOfInterference it(Bloi);
     for(; it.More(); it.Next()) {
       const Handle(TopOpeBRepDS_Interference)& IB = it.Value();
-      TopAbs_ShapeEnum SBB,SAB;Standard_Integer IBB,IAB;TopOpeBRepDS_Kind GTB,STB;Standard_Integer GB,SB;
+      TopAbs_ShapeEnum SBB,SAB;Standard_Integer IBB,IAB;TopOpeBRepDS_Kind GTB = TopOpeBRepDS_UNKNOWN,STB;Standard_Integer GB = 0,SB;
       FDS_Idata(IB,SBB,IBB,SAB,IAB,GTB,GB,STB,SB);
       if (GTB == TopOpeBRepDS_EDGE && GB == SI) {
 	// la face IB1 a une interference dont la geometrie est l'arete SI.
@@ -363,7 +363,7 @@ Standard_EXPORT Standard_Boolean FDS_SIi
     TopOpeBRepDS_ListIteratorOfListOfInterference it(Aloi);
     for(; it.More(); it.Next()) {
       const Handle(TopOpeBRepDS_Interference)& IA = it.Value();
-      TopAbs_ShapeEnum SBA,SAA;Standard_Integer IBA,IAA;TopOpeBRepDS_Kind GTA,STA;Standard_Integer GA,SA;
+      TopAbs_ShapeEnum SBA,SAA;Standard_Integer IBA,IAA;TopOpeBRepDS_Kind GTA = TopOpeBRepDS_UNKNOWN,STA;Standard_Integer GA = 0,SA;
       FDS_Idata(IA,SBA,IBA,SAA,IAA,GTA,GA,STA,SA);
       if (GTA == TopOpeBRepDS_EDGE && GA == SI) {
 	// la face IA1 a une interference dont la geometrie est l'arete IS.
@@ -660,7 +660,7 @@ Standard_EXPORT void FUN_ds_PURGEforE9(c
     TopOpeBRepDS_ListIteratorOfListOfInterference it(l3dF);
     while (it.More()) {
       const Handle(TopOpeBRepDS_Interference)& I = it.Value();
-      TopAbs_ShapeEnum SB,SA;Standard_Integer IB,IA;TopOpeBRepDS_Kind GT,ST;Standard_Integer G,S;
+      TopAbs_ShapeEnum SB,SA;Standard_Integer IB,IA;TopOpeBRepDS_Kind GT,ST;Standard_Integer G,S = 0;
       FDS_Idata(I,SB,IB,SA,IA,GT,G,ST,S);      
       Standard_Boolean FhasGE = FDS_SIisGIofIofSBAofTofI(BDS,IE,I);
       if (FhasGE) {removed = Standard_True; l3dF.Remove(it); continue;} // E has split ON F (cto904A3;e19,f14)
@@ -749,7 +749,7 @@ Standard_EXPORT void FUN_ds_completeforS
 //	const TopOpeBRepDS_Transition& T = I->Transition();
 #endif
 	TopOpeBRepDS_Kind GT,ST;
-        Standard_Integer G1,S;
+        Standard_Integer G1 = 0,S = 0;
         FDS_data(I,GT,G1,ST,S);
 	TopAbs_ShapeEnum tsb,tsa; Standard_Integer isb,isa; FDS_Tdata(I,tsb,isb,tsa,isa);
 	const TopoDS_Edge& ES = TopoDS::Edge(BDS.Shape(S)); 
@@ -869,7 +869,7 @@ Standard_EXPORT void FUN_ds_completeforS
 #ifdef DEB
 //      const TopOpeBRepDS_Transition& T = I->Transition();
 #endif
-      TopOpeBRepDS_Kind ST; Standard_Integer S; FDS_data(I,K,G,ST,S);
+      TopOpeBRepDS_Kind ST; Standard_Integer S = 0; FDS_data(I,K,G,ST,S);
       TopAbs_ShapeEnum tsb,tsa; Standard_Integer isb,isa; FDS_Tdata(I,tsb,isb,tsa,isa);
       const TopoDS_Face& FTRA= TopoDS::Face(BDS.Shape(isb));
       const TopoDS_Edge& ES = TopoDS::Edge(BDS.Shape(S)); 
@@ -995,7 +995,7 @@ static Standard_Boolean FUN_ds_completef
   // Tr
   //---
   const Handle(TopOpeBRepDS_Interference)& I3 = l3d.First();
-  TopOpeBRepDS_Kind K3,ST3; Standard_Integer G3,S3; FDS_data(I3,K3,G3,ST3,S3);
+  TopOpeBRepDS_Kind K3,ST3; Standard_Integer G3,S3 = 0; FDS_data(I3,K3,G3,ST3,S3);
   TopAbs_ShapeEnum tsb3,tsa3; Standard_Integer isb3,isa3; FDS_Tdata(I3,tsb3,isb3,tsa3,isa3);
   IES = S3; ITRASHA = isb3;
   
@@ -1313,7 +1313,7 @@ static Standard_Integer FUN_EisSE2(const
 {
   const TopOpeBRepDS_Transition& T = I->Transition();
   const TopAbs_Orientation O = T.Orientation(TopAbs_IN);
-  TopAbs_ShapeEnum SB,SA;Standard_Integer IB,IA;TopOpeBRepDS_Kind GT,ST;Standard_Integer G,S;
+  TopAbs_ShapeEnum SB,SA;Standard_Integer IB,IA;TopOpeBRepDS_Kind GT = TopOpeBRepDS_UNKNOWN,ST;Standard_Integer G = 0,S;
   FDS_Idata(I,SB,IB,SA,IA,GT,G,ST,S);
   if (GT != TopOpeBRepDS_VERTEX) return NONE;
   if (M_EXTERNAL(O)) return NONE;
@@ -1443,14 +1443,14 @@ Standard_EXPORT Standard_Integer FUN_ds_
   TopOpeBRepDS_ListIteratorOfListOfInterference itE(LIE);
   for (; itE.More(); itE.Next()){
     const Handle(TopOpeBRepDS_Interference)& IE = itE.Value();
-    TopOpeBRepDS_Kind GTE,STE; Standard_Integer GE,SE; FDS_data(IE,GTE,GE,STE,SE);
+    TopOpeBRepDS_Kind GTE,STE; Standard_Integer GE = 0,SE; FDS_data(IE,GTE,GE,STE,SE);
     TopAbs_ShapeEnum tsb,tsa; Standard_Integer isb,isa; FDS_Tdata(IE,tsb,isb,tsa,isa);
     
     TopOpeBRepDS_ListIteratorOfListOfInterference itF(LIF);
     Standard_Boolean is3d = Standard_False;
     for (; itF.More(); itF.Next()){
       const Handle(TopOpeBRepDS_Interference)& IF = itF.Value();
-      TopOpeBRepDS_Kind GTF,STF; Standard_Integer GF,SF; FDS_data(IF,GTF,GF,STF,SF);
+      TopOpeBRepDS_Kind GTF,STF; Standard_Integer GF = 0,SF = 0; FDS_data(IF,GTF,GF,STF,SF);
       if (GE != GF) continue;
       if (SF==isb) {
 	is3d = Standard_True;
@@ -1501,7 +1501,7 @@ Standard_EXPORT void FUN_ds_completeforS
       if (n2 < 1) continue;
 
       const Handle(TopOpeBRepDS_Interference)& I = l2.First();
-      TopOpeBRepDS_Kind GT,ST; Standard_Integer S; FDS_data(I,GT,G,ST,S);
+      TopOpeBRepDS_Kind GT,ST; Standard_Integer S = 0; FDS_data(I,GT,G,ST,S);
       TopAbs_ShapeEnum tsb,tsa; Standard_Integer isb,isa; FDS_Tdata(I,tsb,isb,tsa,isa);
       const TopoDS_Edge& ES = TopoDS::Edge(BDS.Shape(S));
       const TopoDS_Face& FTRA = TopoDS::Face(BDS.Shape(isb));
@@ -1512,7 +1512,7 @@ Standard_EXPORT void FUN_ds_completeforS
       Standard_Boolean hasFOR=Standard_False,hasREV=Standard_False;
       for (TopOpeBRepDS_ListIteratorOfListOfInterference it(l2); it.More(); it.Next()){
 	const Handle(TopOpeBRepDS_Interference)& I2 = it.Value();
-	TopOpeBRepDS_Kind GT2,ST2; Standard_Integer G2,S2; FDS_data(I2,GT2,G2,ST2,S2);
+	TopOpeBRepDS_Kind GT2,ST2; Standard_Integer G2,S2 = 0; FDS_data(I2,GT2,G2,ST2,S2);
 	TopAbs_ShapeEnum tsb2,tsa2; Standard_Integer isb2,isa2; FDS_Tdata(I2,tsb2,isb2,tsa2,isa2);
 	Standard_Boolean error = (S2 != S) || (isb2 != isb);
 	if (error) return; // nyi raise
@@ -2081,8 +2081,8 @@ Standard_EXPORT void FUN_ds_PointToVerte
       Standard_Integer Scur = 0; Standard_Boolean Gfaulty = Standard_False;
       for (TopOpeBRepDS_ListIteratorOfListOfInterference it(loi); it.More(); it.Next()){
 	const Handle(TopOpeBRepDS_Interference)& I = it.Value();
-	TopOpeBRepDS_Kind GT,ST;
-        Standard_Integer G1,S;
+	TopOpeBRepDS_Kind GT,ST = TopOpeBRepDS_UNKNOWN;
+        Standard_Integer G1,S = 0;
         FDS_data(I,GT,G1,ST,S);
 	if (ST != TopOpeBRepDS_EDGE) continue;
 	if (Scur == 0) {
@@ -2142,8 +2142,8 @@ Standard_EXPORT void FUN_ds_PointToVerte
 	if (CPI.IsNull()) continue;
 
 	Standard_Real par = CPI->Parameter(); 
-	TopOpeBRepDS_Kind GT,ST;
-        Standard_Integer G1,S;
+	TopOpeBRepDS_Kind GT,ST = TopOpeBRepDS_UNKNOWN;
+        Standard_Integer G1,S = 0;
         FDS_data(I,GT,G1,ST,S);
 	const TopOpeBRepDS_Transition& T = I->Transition();
 	Standard_Real parvG; Standard_Boolean ok = FUN_tool_parVonE(vG,TopoDS::Edge(s),parvG);
@@ -2172,7 +2172,7 @@ static Standard_Boolean FUN_redusamshaon
 // attached to edge(EIX) : IFOR=(FORWARD(ES),G,ES) + IREV=(REVERSED(ES),G,ES)
 {
   newI.Nullify();
-  TopAbs_ShapeEnum SB,SA;Standard_Integer IB,IA;TopOpeBRepDS_Kind GT,ST;Standard_Integer G,S;
+  TopAbs_ShapeEnum SB,SA;Standard_Integer IB,IA;TopOpeBRepDS_Kind GT = TopOpeBRepDS_UNKNOWN,ST;Standard_Integer G = 0,S = 0;
   FDS_Idata(I,SB,IB,SA,IA,GT,G,ST,S); 
   const TopoDS_Edge& E = TopoDS::Edge(BDS.Shape(EIX));
   Standard_Real parE = FDS_Parameter(I); Standard_Real f,l; FUN_tool_bounds(E,f,l);
@@ -2253,7 +2253,7 @@ Standard_EXPORT void FUN_ds_redusamsha(c
 	  const TopOpeBRepDS_Transition& T1 = I1->Transition();
 	  TopAbs_Orientation O1 = T1.Orientation(TopAbs_IN);
 	  if (!M_FORWARD(O1) && !M_REVERSED(O1)) {it1.Next(); continue;}
-	  TopAbs_ShapeEnum SB1,SA1;Standard_Integer IB1,IA1;TopOpeBRepDS_Kind GT1,ST1;Standard_Integer G1,S1;
+	  TopAbs_ShapeEnum SB1,SA1;Standard_Integer IB1,IA1;TopOpeBRepDS_Kind GT1,ST1;Standard_Integer G1,S1 = 0;
 	  FDS_Idata(I1,SB1,IB1,SA1,IA1,GT1,G1,ST1,S1);
 	  if (IB1 != IA1) {it1.Next(); continue;}
 	  
@@ -2263,7 +2263,7 @@ Standard_EXPORT void FUN_ds_redusamsha(c
 	    const Handle(TopOpeBRepDS_Interference)& I2 = it2.Value();
 	    const TopOpeBRepDS_Transition& T2 = I2->Transition();
 	    TopAbs_Orientation O2 = T2.Orientation(TopAbs_IN);
-	    TopAbs_ShapeEnum SB2,SA2;Standard_Integer IB2,IA2;TopOpeBRepDS_Kind GT2,ST2;Standard_Integer G2,S2;
+	    TopAbs_ShapeEnum SB2,SA2;Standard_Integer IB2,IA2;TopOpeBRepDS_Kind GT2,ST2;Standard_Integer G2,S2 = 0;
 	    FDS_Idata(I2,SB2,IB2,SA2,IA2,GT2,G2,ST2,S2);
 	    if (IB2 != IA2) {it2.Next(); continue;}
 	    if (IB1 != IB2) {it2.Next(); continue;} // same fTRASHA
@@ -2312,7 +2312,7 @@ Standard_EXPORT Standard_Boolean FUN_ds_
   for (TopOpeBRepDS_ListIteratorOfListOfInterference it(LI); it.More(); it.Next()){
     const Handle(TopOpeBRepDS_Interference)& I = it.Value();
     const TopOpeBRepDS_Transition& T = I->Transition();
-    TopOpeBRepDS_Kind GT,ST; Standard_Integer G,S; FDS_data(I,GT,G,ST,S);
+    TopOpeBRepDS_Kind GT,ST; Standard_Integer G = 0,S; FDS_data(I,GT,G,ST,S);
     Standard_Boolean found = (G == GI) && (T.Index() == ITRA);
     if (found) return Standard_True;
   }
Index: oce/src/XCAFPrs/XCAFPrs_AISObject.cxx
===================================================================
--- oce.orig/src/XCAFPrs/XCAFPrs_AISObject.cxx
+++ oce/src/XCAFPrs/XCAFPrs_AISObject.cxx
@@ -252,10 +252,10 @@ void XCAFPrs_AISObject::AddStyledItem (c
   }
   case 1:
     {
-      Standard_Real prevangle ;
-      Standard_Real newangle  ; 
-      Standard_Real prevcoeff ;
-      Standard_Real newcoeff  ; 
+      Standard_Real prevangle = 0.0;
+      Standard_Real newangle = 0.0  ; 
+      Standard_Real prevcoeff = 0.0 ;
+      Standard_Real newcoeff = 0.0 ; 
       
       Standard_Boolean isOwnDeviationAngle = OwnDeviationAngle(newangle,prevangle);
       Standard_Boolean isOwnDeviationCoefficient = OwnDeviationCoefficient(newcoeff,prevcoeff);
Index: oce/src/math/math_DirectPolynomialRoots.cxx
===================================================================
--- oce.orig/src/math/math_DirectPolynomialRoots.cxx
+++ oce/src/math/math_DirectPolynomialRoots.cxx
@@ -62,6 +62,8 @@ const Standard_Real ZERO = 1.0e-30;
         Standard_Real Sol = IniSol;
         Standard_Real IniVal = Value(N, Poly, IniSol);
         Standard_Integer Index;
+        // set a possible initial value to avoid compiler complaining
+        Val = IniVal;
 
 //      cout << "Improve\n";
         for(Index = 1; Index < 10; Index++) {
